api/v1/hostdefinition_types.go

```diff
@@ -0,0 +1,89 @@
+/**
+ * Copyright 2019 IBM Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// HostDefinitionSpec defines the desired state of HostDefinition
+type HostDefinitionSpec struct {
+	HostDefinition IBMBlockCSIHostDefinitionSpec `json:"hostDefinition"`
+}
+
+// IBMBlockCSIHostDefinitionSpec defines the observed state of HostDefinition
+type IBMBlockCSIHostDefinitionSpec struct {
+	StorageServer     string `json:"storageServer"`
+	HostNameInStorage string `json:"hostNameInStorage"`
+	SecretName        string `json:"secretName"`
+	SecretNamespace   string `json:"secretNamespace"`
+
+	// +kubebuilder:validation:Optional
+	Message string `json:"message"`
+	// +kubebuilder:validation:Optional
+	ConnectvityType string `json:"connectvityType"`
+	// +kubebuilder:validation:Optional
+	ConnectivityPorts string `json:"connectivityPorts"`
```

Comment:
shouldn't be a list?
```suggestion
	ConnectivityPorts []string `json:"connectivityPorts"`
```

Reply:
this field is removed for now